static/usr/lib/core/rtc-sys-time-init

```diff
@@ -0,0 +1,105 @@
+#!/bin/sh
+
+# Adjust system time for the following reasons:
+#     1) Sync system time to RTC time
+#          - On older kernel versions, modules with HTCOSYS enabled cannot sync system time with
+#            RTC (unlike built-in RTC drivers)
+#          - Where it can sync at driver level, this might cause harmless re-sync
+#     2) Correct driver/module system time sync to invalid time (earlier than most recent time stamp)
+#          - Built-in/loadable RTC driver with HCTOSYS enabled may set invalid time due to uninitialized,
+#            invalid time e.g. caused by bad battery
+#
+# If the RTC time is not valid, or cannot be set, forward the system time to most recent timestamp
+
+# Expects single argument that is service instance name which is
+# based on the associated RTC device unit name in the form dev-rtc*
+dev_unit="$1"
+
```

Comment:
I feel like we should look at hctosys in the devpath, if it set to 1, we do not have to do anything here. That way we know we do not have conflict with the other script.

Reply:
Ah but the other script is not called because of `IMPORT{cmdline}="rtc_sys_time_init", ENV{rtc_sys_time_init}!="?*"`.

Reply:
Yes, the generator and udev triggers are mutually exclusive. Did incorporate looking for `hctosys` value in this script to prevent unnecessary re-sync for system time to RTC if driver already done it, as per your other suggestion.